includes/payment-methods/class-upe-payment-gateway.php

```diff
@@ -46,6 +46,14 @@ class UPE_Payment_Gateway extends WC_Payment_Gateway_WCPay {
 
 	const UPE_APPEARANCE_TRANSIENT = 'wcpay_upe_appearance';
 
+	const COOKIE_UPE_PAYMENT_INTENT = 'wcpay_upe_payment_intent';
+
+	const COOKIE_UPE_SETUP_INTENT = 'wcpay_upe_setup_intent';
+
+	const COOKIE_UPE_EXPIRES_IN_MINS = 10;
```

Comment:
10 mins only, any concerns increasing this value? Perhaps a couple of hours?

Reply:
My concern is noted in 7c20418. In a nutshell, an over-lived cookie may cause undesired side effects, [like this](https://github.com/Automattic/woocommerce-payments/pull/3684#pullrequestreview-858935708):

>  If you force a transaction to fail and then return to the checkout, something a little interesting happens: the UPE loads, but it only shows inputs for the previous payment method used and it is now impossible to change this!

I have tried to cover as many failure scenarios as I could think of, with help from @FangedParakeet of course. But in case an edge case got missed, a lower TTL would help mitigate side effects.